Update f5_tmsh_ssh.py

Trasmontinho opened the pull request:
Extract from F5 Website (https://support.f5.com/csp/article/K12029):
```
Users can enter tmsh in two ways:

If you are granted Advanced shell (bash) access, you must type tmsh in to the command line after logging in to the BIG-IP system.
If you are granted tmsh access, you enter tmsh directly when logging in to the BIG-IP system.


```


Starting tests...good luck:
================================================================================================ test session starts =================================================================================================
platform linux -- Python 3.6.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/trasmontinho/tests_f5/tests
collected 22 items                                                                                                                                                                                                   

test_netmiko_show.py::test_disable_paging PASSED                                                                                                                                                               [  4%]
test_netmiko_show.py::test_terminal_width PASSED                                                                                                                                                               [  9%]
test_netmiko_show.py::test_ssh_connect PASSED                                                                                                                                                                  [ 13%]
test_netmiko_show.py::test_ssh_connect_cm PASSED                                                                                                                                                               [ 18%]
test_netmiko_show.py::test_send_command_timing PASSED                                                                                                                                                          [ 22%]
test_netmiko_show.py::test_send_command_timing_no_cmd_verify PASSED                                                                                                                                            [ 27%]
test_netmiko_show.py::test_send_command PASSED                                                                                                                                                                 [ 31%]
test_netmiko_show.py::test_send_command_no_cmd_verify PASSED                                                                                                                                                   [ 36%]
test_netmiko_show.py::test_cmd_verify_decorator PASSED                                                                                                                                                         [ 40%]
test_netmiko_show.py::test_send_command_global_cmd_verify PASSED                                                                                                                                               [ 45%]
test_netmiko_show.py::test_complete_on_space_disabled SKIPPED                                                                                                                                                  [ 50%]
test_netmiko_show.py::test_send_command_textfsm SKIPPED (TextFSM/ntc-templates not supported on this platform)                                                                                                 [ 54%]
test_netmiko_show.py::test_send_command_ttp SKIPPED (TTP template not existing for this platform)                                                                                                              [ 59%]
test_netmiko_show.py::test_send_command_genie SKIPPED (Genie not supported on this platform)                                                                                                                   [ 63%]
test_netmiko_show.py::test_base_prompt PASSED                                                                                                                                                                  [ 68%]
test_netmiko_show.py::test_strip_prompt PASSED                                                                                                                                                                 [ 72%]
test_netmiko_show.py::test_strip_command PASSED                                                                                                                                                                [ 77%]
test_netmiko_show.py::test_normalize_linefeeds PASSED                                                                                                                                                          [ 81%]
test_netmiko_show.py::test_clear_buffer PASSED                                                                                                                                                                 [ 86%]
test_netmiko_show.py::test_enable_mode PASSED                                                                                                                                                                  [ 90%]
test_netmiko_show.py::test_disconnect PASSED                                                                                                                                                                   [ 95%]
test_netmiko_show.py::test_disconnect_no_enable PASSED                                                                                                                                                         [100%]

====================================================================================== 18 passed, 4 skipped in 64.90s (0:01:04) ======================================================================================
================================================================================================ test session starts =================================================================================================
platform linux -- Python 3.6.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/trasmontinho/tests_f5/tests
collected 9 items                                                                                                                                                                                                    

test_netmiko_config.py::test_ssh_connect PASSED                                                                                                                                                                [ 11%]
test_netmiko_config.py::test_enable_mode PASSED                                                                                                                                                                [ 22%]
test_netmiko_config.py::test_config_mode PASSED                                                                                                                                                                [ 33%]
test_netmiko_config.py::test_exit_config_mode PASSED                                                                                                                                                           [ 44%]
test_netmiko_config.py::test_config_set PASSED                                                                                                                                                                 [ 55%]
test_netmiko_config.py::test_config_set_longcommand PASSED                                                                                                                                                     [ 66%]
test_netmiko_config.py::test_config_hostname PASSED                                                                                                                                                            [ 77%]
test_netmiko_config.py::test_config_from_file PASSED                                                                                                                                                           [ 88%]
test_netmiko_config.py::test_disconnect PASSED                                                                                                                                                                 [100%]

================================================================================================= 9 passed in 14.49s =================================================================================================

ktbyers commented on netmiko/f5/f5_tmsh_ssh.py:
```diff
@@ -25,3 +25,15 @@ def tmsh_mode(self, delay_factor=1):
+
+    def check_config_mode(self, check_string=") #", pattern=""):
+        """Checks if the device is in configuration mode or not."""
+        return super().check_config_mode(check_string=check_string, pattern=pattern)
```
I think I am going to change this to just `return True` unless you see an issue with that? That is what I typically do for platforms that have no separate config mode.

Everything else looks good on this.

Trasmontinho replied:
Ok no problem !

Trasmontinho replied:
Ok I've tried with `return True` in place of  `return super().check_config_mode(check_string=check_string, pattern=pattern)`

and the test_netmiko_config.py drops me the error below : 

```
    def test_exit_config_mode(net_connect, commands, expected_responses):
        """
        Test exit config mode
        """
        net_connect.exit_config_mode()
>       assert net_connect.check_config_mode() is False
E       assert True is False
E        +  where True = <bound method F5TmshSSH.check_config_mode of <netmiko.f5.f5_tmsh_ssh.F5TmshSSH object at 0x7fb26f50aef0>>()
E        +    where <bound method F5TmshSSH.check_config_mode of <netmiko.f5.f5_tmsh_ssh.F5TmshSSH object at 0x7fb26f50aef0>> = <netmiko.f5.f5_tmsh_ssh.F5TmshSSH object at 0x7fb26f50aef0>.check_config_mode

test_netmiko_config.py:42: AssertionError
================================================================================================ short test summary info ================================================================================================
FAILED test_netmiko_config.py::test_exit_config_mode - assert True is False
============================================================================================= 1 failed, 8 passed in 12.28s ==============================================================================================
```

Trasmontinho replied:
I'll add probably later commit stuff because when you modify some config and you have F5 loadbalancer in cluster, 

the prompt is as below before the change : 
`trasmontinho@(f5a)(cfg-sync Standalone)(/S1-green-P::Active)(/Common)(tmos)#`
or 
`trasmontinho@(f5b)(cfg-sync Standalone)(/S1-green-P::Standby)(/Common)(tmos)#`

As soon as you config some data the prompt change to  : 
`trasmontinho@(f5b)(cfg-sync Changes Pending)(/S1-green-P::Standby)(/Common)(tmos)#`
or 
`trasmontinho@(f5a)(cfg-sync Changes Pending)(/S1-green-P::Standby)(/Common)(tmos)#`

ktbyers replied:
Okay, that test error is fine. It is a problem with the test which I will correct.

Trasmontinho replied:
Ok maybe I'll drop you an issue on it. 
I've got to fill the` enable_prompt` to pass the test. If I don't, I've got a `KeyError 'enable_prompt'`
Maybe such stuff should be warn you on  missing tests if those elements are not present.

ktbyers replied:
Yeah, just put that as the same as the standard prompt on  your device so whatever the full prompt is after logging into `tmos` for the device you are testing.

I fixed the `exit_config_mode` test issue in new Pull-Request (#2153)

Trasmontinho commented:
note for later as those information are in the prompt (see the previous example) : 


Green | None. | This status indicates one of the following conditions:The device has the most recent set of configuration data within the device group.The device is a standalone device or is the only device group member.Recommended action: None.
-- | -- | --
Blue | Awaiting Initial Sync | This status indicates one of the following conditions:The device has no configuration changes to be synced since joining the device group.The device is member of a device group with autosync enabled, and no changes have been made on any device in the device group.The device has not yet received a sync from another device and has no configuration changes to be synced to other members of the device group.Recommended action: Perform the appropriate type of config sync.
Yellow | Changes Pending | The device has recent configuration changes since the last sync that have not yet been synchronized to the other members of the device group.Recommended action: Sync the device with the most recent configuration to the other members of the device group.
Red | Changes Pending | A device group member reported in its most recent status update that it had changes pending but is now disconnected from the local device.Recommended action: When the disconnected device comes back online, sync the device with the most recent changes to the device group.
Yellow | Awaiting Initial Sync with Changes Pending | This status indicates one of the following conditions:The configuration on the device has changed since the device joined the device group.Recommended action: Sync the device to the device group.The device has not yet received a sync from another device but has configuration changes to be synced to other members of the device group.Recommended action: Sync the device with the most recent configuration to this device.
Red | Does not have the last synced configuration, and has changes pending | The device received at least one sync previously but did not receive the last synced configuration, and the configuration on the device has changed since the last sync.Recommended action: Sync the device with the most recent configuration to this device.
Red | Disconnected | The local device does not recognize the disconnected device.Recommended actions:View the screens at Device Management > Device Trust to see if the disconnected device is a member of the local trust domain, and if not, add the device to the domain.Use the screen at Device ManagementDevices to view the specified config sync address of the disconnected device and determine whether the local device has a route to that address.
Red | Device does not recognize membership in this group | The device does not recognize that it is a member of the device group.Recommended action: Log into the relevant device and view the screens at Device Management > Device Groups to see if the device is a member of the device group. If not, add the device to the device group.
Red | No config sync address has been specified for this device. | The device does not have a config sync address.Recommended action: Log into the relevant device, and using the screen at Device Management > Devices, specify the IP address that you want remote devices to use to sync configuration data to the device. As a best practice, this address should be a non-floating self IP address associated with an internal VLAN. The address must either be on the same subnet as the other devices in the device group or have a route to that address defined on the other devices.
Red | Does not have the last synced configuration | The device previously received the configuration from other members of the device group but did not receive the last synced configuration.Recommended action: Sync the device group to the device.

ktbyers commented:
Superseded by:

https://github.com/ktbyers/netmiko/pull/2153
